pkgs/development/compilers/llvm/8/default.nix

```diff
@@ -78,48 +78,78 @@ let
 
     lldb = callPackage ./lldb.nix {};
 
+    # Below, is the LLVM bootstrapping logic. It handles building a
+    # fully LLVM toolchain from scratch. No GCC toolchain should be
+    # pulled in. As a consequence, it is very quick to build different
+    # targets provided by LLVM and we can also build for what GCC
+    # doesn’t support like LLVM. Probably we should move to some other
+    # file.
```

Comment:
If we have `libc`, `libcxx`, `librt`, `libunwind`` attributes in cc-wrapper, I think most of this logic could move there even though some it is clang-specific.

Reply:
You would still need the 4 ccs though right?

Reply:
Perhaps but it could arise organically like:
```nix
// all-packages.nix
libc = ...;
librt = ...;
```
```
// llvm default.nix
xxx = ... (stdenvNoLlibs.override {
  foo = libraries.noNull; 
}) ...;
```